src/pages/search/registration_search/filters/RegistrationFacetsFilter.tsx

```diff
@@ -76,23 +78,30 @@ export const RegistrationFacetsFilter = ({ registrationQuery }: Pick<SearchPageP
       )}
 
       {topLevelOrganizationFacet.length > 0 && (
-        <FacetItem title={t('common.institution')} dataTestId={dataTestId.aggregations.institutionFacets}>
+        <FacetItem
+          title={t('common.institution')}
+          dataTestId={dataTestId.aggregations.institutionFacets}
+          renderCustomSelect={
+            <SearchForInstitutionFacetItem
+              onSelectInstitution={(identifier) => addFacetFilter(ResultParam.TopLevelOrganization, identifier)}
+            />
+          }>
           {topLevelOrganizationFacet.map((facet) => {
-            const isSelected = !!registrationParams.topLevelOrganization?.includes(facet.key);
+            const institutionIdentifier = getIdentifierFromId(facet.key);
+            const isSelected = !!registrationParams.topLevelOrganization?.includes(institutionIdentifier);
 
             return (
               <FacetListItem
                 key={facet.key}
-                identifier={facet.key}
-                dataTestId={dataTestId.aggregations.facetItem(facet.key)}
+                dataTestId={dataTestId.aggregations.facetItem(institutionIdentifier)}
```

Comment:
Liten endring for data-testid her, @Eiriknil. Vil nå være postfikset med bare identifier i stedet for hele URI til institusjonen